apps/212-daterangepicker/app.R

```diff
@@ -0,0 +1,143 @@
+      });
+
+      jst.add(Jster.shiny.waitUntilStable);
+      jst.add(function() {
+        var slider_val = $('#slider_val').text().trim() - 0; //convert string to number by adding 0
+        var up_from = $('#up_from').text().trim() - 0;
+        var up_to = $('#up_to').text().trim() - 0;
+
+        // Verify from and to date values match accordingly with the slider value
+        Jster.assert.isEqual(slider_val, date_info.slider_val, {slider_value: slider_val});
+        Jster.assert.isEqual(up_from, date_info.slider_val - 1, {from: up_from});
+        Jster.assert.isEqual(up_to, date_info.slider_val + 1, {to: up_to});
+
+        // Verify from and to dates (yyyy-mm-dd format) in the date picker text box matches with the expected
+        var from_date_value = '' + (date_info.slider_val - 1);
+        if (from_date_value.length == 1) from_date_value = '0' + from_date_value;
+        Jster.assert.isEqual(Jster.daterangepicker.from.value('inDateRange'), '2013-05-' + from_date_value)
+
+        var to_date_value = '' + (date_info.slider_val + 1);
+        if (to_date_value.length == 1) to_date_value = '0' + to_date_value;
+        Jster.assert.isEqual(Jster.daterangepicker.to.value('inDateRange'), '2013-05-' + to_date_value)
+      })
+
+      // From Datepicker: Possible dates check
+      jst.add(function() {
+        Jster.daterangepicker.from.click('inDateRange');
+      })
+      jst.add(function() {
+        var possible_values = $('#possible').text().trim();
+        possible_values = JSON.parse(possible_values)
+
+        console.log('Possible dates:' , Jster.datepicker.possibleDates('inDateRange'))
+        Jster.assert.isEqual(
+          Jster.datepicker.possibleDates('inDateRange'), date_info.possible_dates
+        );
+        Jster.daterangepicker.from.bs.hide('inDateRange');
+      })
+
+      // To Datepicker: Possible dates check
+      jst.add(function() {
+        Jster.daterangepicker.to.click('inDateRange');
+      })
+      jst.add(function() {
+        var possible_values = $('#possible').text().trim();
+        possible_values = JSON.parse(possible_values)
+
+        console.log('Possible dates:' , Jster.datepicker.possibleDates('inDateRange'))
+        Jster.assert.isEqual(
+          Jster.datepicker.possibleDates('inDateRange'), date_info.possible_dates
+        );
+        Jster.daterangepicker.to.bs.hide('inDateRange');
+      })
+
+    })
+    jst.test();
+   ")
+)
+
+# SERVER LOGIC
+server <- function(input, output, session) {
+
+  # include shinyjster server call at top of server definition
+  shinyjster::shinyjster_server(input, output)
+
+  observe({
+    date <- as.Date(paste0("2013-05-", input$n))
+
+    updateDateRangeInput(session, "inDateRange",
+                         label = "Date range picker",
+                         start = date - 1,
+                         end = date + 1,
+                         min = date - 5,
+                         max = date + 5
+    )
+  }) %>%
+    bindEvent(input$n)
```

Comment:
I don't think this `bindEvent` is necessary, since the observer's only reactive input is `input$n`.